src/sumBy.test-d.ts

```diff
@@ -0,0 +1,85 @@
+import { sumBy } from "./sumBy";
+import { expectTypeOf } from "vitest";
+import { pipe } from "./pipe";
+
+test("empty array", () => {
+  const result1 = sumBy([], BigInt);
+  expectTypeOf(result1).toEqualTypeOf<0>();
+
+  const result2 = sumBy([], toNumber);
+  expectTypeOf(result2).toEqualTypeOf<number>();
```

Comment:
To be consistent, the typing of `sumBy` for `number` should also be fixed. This should also return a constant `0`, just like for `bigint`.

Reply:
The result of the function `sumBy` has been refactored for stricter, please check it again